src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldSymbol.cs

```diff
@@ -211,33 +211,19 @@ public sealed override string Name
         }
 
         internal override LexicalSortKey GetLexicalSortKey()
-        {
-            return new LexicalSortKey(_location, this.DeclaringCompilation);
-        }
+            => new LexicalSortKey(_syntaxReference, this.DeclaringCompilation);
```

Comment:
I wonder if we should think about deleting the `Location, Compilation` constructor, and force instanciations to use SyntaxRefs directly. Maybe a followup?